### DISCUSSION TITLE
NX8-E is not working / NX-584 is working

### DISCUSSION THREAD

**Discussion by raymonvdm** [tjko/nxgipd#20] [General]
I`m trying to connect to a NX8-E panel but without any succes so far. However when i connect to an NX-584 (Home Automation) the connection is fine but i get the following warning

Panel Model: Unknown (Panel ID=2). Please report your alarm panel model and id to nxgipd developers.

Which ID do i need to lookup? The panel is an NX-8 with an NX-584

**Comment by tjko** [comment]
So using the serial port on the alarm panel main board doesn't work, but using serial port on NX-584 module works?    Have you verified pinout on the adapter cable you're using to connect to the 10-pin IDC connector on the panel itself? (https://github.com/tjko/nxgipd/wiki/NX-8E-Serial-Connector-Pinout), also is the internal serial port enabled, I seem to recall that the on-board serial port is not enabled by default.

Can you send log from nxgipd when it starts (generated with logging level set to 2 or 3), it might shed some light what is going on...

So far "NX-8E" panels I've seen report Panel ID=4 (sold in North America), so maybe you have different version/variant... Photo of the alarm panel main board might help identify it...

**Comment by raymonvdm** [comment]
I might need to do some clarification

**Unit1** This is a NX8 alarm panel sold in the EU (Panel ID=2) and this panel is equiped with an NX-584 do to the missing onboard serial port. I have swapped jumpers on the NX-584 to verify the cables i have in use. I also verified the working of the RaspberryPi and USB converter running nxgipd. This panel is working for two years using an serial connection to my HomeSeer system using other CADDX software tool

**Unit2** This is a NX8plus alarm panel sold in the EU and this panel has the onboard RS232, which is enabled by the installer and set to 9600Baud / ASCII ( 10-pin IDC cable is also supplied by the Installer) I cannot open this panel without alarm bells ringing so trouble shooting is a little bit of a hassle

**Unit3** I`m waiting for this unit to appear in my test lab :-) So i can test out the IDC cable

**Comment by raymonvdm** [comment]
I just tried again with **Unit2** and got this response

root@raspberrypi:/home/nxgipd# ./nxgipd 
Loading configuration...
Opening serial port: /dev/ttyUSB0
Establishing communications...
Detecting panel model...
Detected panel: NX-8E (maxzones=192, maxpartitions=8)
Interface Firmware version v5.38 detected
Program started: nxgipd v1.1.1beta (2016-11-14)
Getting system status...
Panel Model: NX-8E
Partition 1 Active
Partition 2 Active
Partition 3 Active
Partition 4 Active
Partition 5 Active
Partition 6 Active
Partition 7 Active
Partition 8 Active
PIN required for local download
AC power ON
50 Hz line power detected
Querying partition statuses...
Partition 1 status change: Ready
Partition 3 status change: Armed, Ready
Partition 4 status change: Ready
Partition 6 status change: Ready, Chime Mode On
Partition 1 status change: Sensor Battery Low, Keyswitch Armed, Last User = 073
Partition 2 status change: Sensor Battery Low, Last User = 073
Partition 3 status change: Sensor Battery Low, Last User = <None>
Partition 4 status change: Sensor Battery Low, Keyswitch Armed, Last User = 099
Partition 5 status change: Sensor Battery Low, Last User = 001
Partition 6 status change: Sensor Battery Low, Keyswitch Armed, Last User = 073
Partition 7 status change: Sensor Battery Low, Last User = 011
Partition 8 status change: Sensor Battery Low, Last User = 011
Querying zone names and statuses...
.........
failed to get system status: -5

failed to get system status: -1
Partition 1 Active
Partition 2 Active
Partition 3 Active
Partition 4 Active
Partition 5 Active
Partition 6 Active
Partition 7 Active
Partition 8 Active
Querying partition statuses...
Partition 2 status change: Ready
Querying zone names and statuses...
..........
failed to get system status: -5
Partition 1 Active
Partition 2 Active
Partition 3 Active
Partition 4 Active
Partition 5 Active
Partition 6 Active
Partition 7 Active
Partition 8 Active
Querying partition statuses...
Querying zone names and statuses...
......................................
failed to get system status: -5
nxgipd: communication problem, giving up
root@raspberrypi:/home/nxgipd#

**Comment by tjko** [comment]
Does this system have any NX-148 (LCD keypads)?  Is querying zone names working (its able to pull zone names and doesnt just generate dummy "Zone xx" names for each zone?

That "failed to get system status: -5" points to zone status request message (command) failing for some reason. Are you able to check system configuration whether that command is enabled?

Alternatively, is this perhaps large alarm system that is rather active? I think the initial code to check system status is not very good handling busy system generating lot of status messages on its own...

**Comment by milwen2005** [comment]
I am having the same or similar problem.  I have an NX-8E. There is an NX-184E keypad installed in the system. The zone status request feature is enabled. Below is my log output.

> 
2017-09-06 21:03:08: Program started: nxgipd v1.1.0 (2017-09-04)
2017-09-06 21:03:08: NX-584 Firmware version v5.38
2017-09-06 21:03:08: Getting system status...
2017-09-06 21:03:08: Panel Model: NX-8E
2017-09-06 21:03:08: Partition 1 Active
2017-09-06 21:03:08: Partition 2 Disabled
2017-09-06 21:03:08: Partition 3 Disabled
2017-09-06 21:03:08: Partition 4 Disabled
2017-09-06 21:03:08: Partition 5 Disabled
2017-09-06 21:03:08: Partition 6 Disabled
2017-09-06 21:03:08: Partition 7 Disabled
2017-09-06 21:03:08: Partition 8 Disabled
2017-09-06 21:03:08: AC power ON
2017-09-06 21:03:08: Querying partition statuses...
2017-09-06 21:03:08: Partition 1 status change: Ready
2017-09-06 21:03:08: Partition 1 status change: Sensor Battery Low, Last User = 001
2017-09-06 21:03:08: Querying zone names and statuses...
2017-09-06 21:03:09: failed to get name for zone 1 (no NX-148E present?)

**Comment by tjko** [comment]
Is the LCD keypad your first keypad (ie. using address 192)?  If not, check that panel programming location 212 is configured correctly...

--

LOCATION 212 - PROGRAMMING THE LCD KEYPAD ADDRESS (1 segment of numerical data)
Certain commands in the NX-8E require it to know the location of at least 1 LCD keypad (if one exists in the system). If your system has an LCD keypad it is recommended that it be placed in partition 1 keypad 1. This will allow location 212 to be left at the factory default. If the LCD keypad is selected as something other than partition 1/ keypad 1 program the appropriate address in location 212. Select the address from the following chart.

KEYPAD|PART 1|PART 2|PART 3|PART 4|PART 5|PART 6|PART 7|PART 8
-- | -- | -- | -- | -- | -- | -- | -- | --
1 | 192 | 193 | 194 | 195 | 196 | 197 | 198 | 199
2 | 200 | 201 | 202 | 203 | 204 | 205 | 206 | 207
3 | 208 | 209 | 210 | 211 | 212 | 213 | 214 | 215
4 | 216 | 217 | 218 | 219 | 220 | 221 | 222 | 223
5 | 224 | 225 | 226 | 227 | 228 | 229 | 230 | 231
6 | 232 | 233 | 234 | 235 | 236 | 237 | 238 | 239
7 | 240 | 241 | 242 | 243 | 244 | 245 | 246 | 247
8 | 248 | 249 | 250 | 251 | 252 | 253 | 254 | 255
